rclpy/test/test_node.py

```diff
@@ -171,6 +178,90 @@ def test_node_names_and_namespaces(self):
         # test that it doesn't raise
         self.node.get_node_names_and_namespaces()
 
+    def assert_qos_equal(self, qos_profile, qos_dict):
+        # Depth and history are skipped because they are not retrieved.
+        self.assertEqual(qos_profile.durability, qos_dict['durability'],
+                         'Durability is unequal')
+        self.assertEqual(qos_profile.reliability, qos_dict['reliability'],
+                         'Reliability is unequal')
+        self.assertEqual(qos_profile.lifespan, qos_dict['lifespan'],
+                         'lifespan is unequal')
+        self.assertEqual(qos_profile.deadline, qos_dict['deadline'],
+                         'Deadline is unequal')
+        self.assertEqual(qos_profile.liveliness, qos_dict['liveliness'],
+                         'liveliness is unequal')
+        self.assertEqual(qos_profile.liveliness_lease_duration,
+                         qos_dict['liveliness_lease_duration'],
+                         'liveliness_lease_duration is unequal')
+
+    @unittest.skipIf(get_rmw_implementation_identifier() != 'rmw_fastrtps_cpp',
```

Comment:
Skipping this test if the implementation is not `rmw_fastrtps_cpp`; should be changed once the other vendors have implemented it as well.